cmd/appgw-ingress/main.go

```diff
@@ -93,13 +94,13 @@ func main() {
 		AppGwName:      env.AppGwName,
 	}
 
-	kubeClient, err := kubernetes.NewForConfig(getKubeClientConfig())
-	if err != nil {
-		glog.Fatal("Error creating Kubernetes client: ", err)
-	}
+	apiConfig := getKubeClientConfig()
+	kubeClient := kubernetes.NewForConfigOrDie(apiConfig)
+	crdClient := versioned.NewForConfigOrDie(apiConfig)
 	namespaces := getNamespacesToWatch(env.WatchNamespace)
 	validateNamespaces(namespaces, kubeClient) // side-effect: will panic on non-existent namespace
 	glog.Info("Ingress Controller will observe the following namespaces:", strings.Join(namespaces, ","))
+	k8sContext := k8scontext.NewContext(kubeClient, namespaces, *resyncPeriod, crdClient)
```

Comment:
should the crdClient be before namespaces ?

Reply:
We could move it a few lines down to ensure namespaces validation passes.

Reply:
I meant that param `crdClient` should passed as a second argument.

Reply:
Sure - that'd be a good change.